deps/rabbit/src/rabbit_mnesia.erl

```diff
@@ -274,6 +162,34 @@ join_cluster(DiscoveryNode, NodeType) ->
             end
     end.
 
+-spec join_cluster
+([node()], rabbit_db_cluster:node_type()) ->
+    ok | {error, any()};
+(node(), rabbit_db_cluster:node_type()) ->
+    ok | {ok, already_member} | {error, {inconsistent_cluster, string()}}.
+
+join_cluster(ClusterNodes, NodeType) when is_list(ClusterNodes) ->
+    %% Join the cluster
+    rabbit_log:info("Clustering with ~tp as ~tp node",
+                    [ClusterNodes, NodeType]),
+    ok = init_db_with_mnesia(ClusterNodes, NodeType,
+                             true, true, _Retry = true),
+    rabbit_node_monitor:notify_joined_cluster(),
+    ok;
+join_cluster(DiscoveryNode, NodeType) when is_atom(DiscoveryNode) ->
+    %% Code to remain compatible `change_cluster_node_type/1' and with older
+    %% CLI.
+    case can_join_cluster(DiscoveryNode) of
+        {ok, ClusterNodes} when is_list(ClusterNodes) ->
+            ok = reset_gracefully(),
```

Comment:
Isn't it already reset in `rabbit_db:join`?

Reply:
The second clause of `rabbit_mnesia:join_cluster()` will only be used by an old CLI (which won't call `rabbit_db` at all). It's unused in the new code path.